### PR TITLE
Support tags on embedded structs

### PR DESCRIPTION
**Pull request by johnrengelman**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea/
```
Comment (nathanielc) [inline .gitignore]: This kind of thing belongs in a user specific .gitignore file http://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file

**Comment by nathanielc** [comment]
@johnrengelman Also, I don't believe the package is actually used by any projects ATM. Am I mistaken?

File: .gitignore (hunk 1 above)
Comment (johnrengelman) [inline .gitignore]: Removed that commit.

**Comment by johnrengelman** [comment]
@nathanielc It's used by `telegraf` - https://github.com/influxdata/telegraf/blob/master/internal/config/config.go#L25

**Comment by nathanielc** [comment]
Awesome, I didn't know Telegraf had migrated. In that case @sparrc Does this look good to you?

**Comment by sparrc** [comment]
Looks fairly innocuous, @johnrengelman can you explain the use-case?

**Comment by johnrengelman** [comment]
@sparrc is support of https://github.com/influxdata/telegraf/pull/1155
In that PR, I implemented a central `struct` and methods for configuring AWS credentials for the various input/output plugins.
While testing I found that if I specified a tag on a field and it's struct was embedded into another struct, then the tag wouldn't be evaluated.

So I had something like:
```
type AWS struct {
  RoleARN  string  "toml:`role_arn`"
}

type Cloudwach struct {
  AWS
}
```

and a config file of:
```
[[inputs.cloudwatch]]
  role_arn = "foo/bar"
```

and that would cause an error saying that it didn't know what to map `role_arn` to.

I worked around it in the above PR, but changing the property to be `RoleArn` which then allowed the default snake-casing to kick in and populate the field, but the behavior is unexpected.

**Comment by sparrc** [comment]
@johnrengelman let's discuss in influxdata/telegraf#1155, I think there are other ways to implement this that are more explicit.

**Comment by johnrengelman** [comment]
@sparrc sounds good. I leave this PR up to you then :)

**Pull request closed without merging**